scripts/prepare_release.sh

```diff
@@ -1,70 +1,38 @@
-    exit
-fi
-
-unrelease= # if non-empty, we're in undo-release mode
-while getopts u OPTLET; do
-    case $OPTLET in
-        u) unrelease=1;;
-        \?)
-            echo 1>&2 "$0: unknown option: -$OPTLET"
-            echo 1>&2 "Try '$0 --help' for more information."
-            exit 3;;
-    esac
-done
-
+# prepare_release.sh — Prepare the source tree for a release.
+#
+# This script switches the repo into “release” mode:
+#   - Updates all tracked `.gitignore` files to stop
+#     ignoring the automatically-generated files.
+#   - Sets the CMake option `GEN_FILES` to OFF to explicitely disable
+#     recreating the automatically-generated files.
+#.  - The script will recursively update the tf-psa-crypto files too.
 
 
-#### .gitignore processing ####
+set -eu
 
-GITIGNORES=$(find . -name ".gitignore")
-for GITIGNORE in $GITIGNORES; do
-    if [ -n "$unrelease" ]; then
-        sed -i '/###START_COMMENTED_GENERATED_FILES###/,/###END_COMMENTED_GENERATED_FILES###/s/^#//' $GITIGNORE
-        sed -i 's/###START_COMMENTED_GENERATED_FILES###/###START_GENERATED_FILES###/' $GITIGNORE
-        sed -i 's/###END_COMMENTED_GENERATED_FILES###/###END_GENERATED_FILES###/' $GITIGNORE
+# Portable inline sed. Helper function that will automatically pre-pend
+# an empty string as the backup suffix (required by macOS sed).
+psed() {
+    # macOS sed does not offer a version
+    if sed --version >/dev/null 2>&1; then
+        sed -i "$@"
+    # macOS/BSD sed
     else
-        sed -i '/###START_GENERATED_FILES###/,/###END_GENERATED_FILES###/s/^/#/' $GITIGNORE
-        sed -i 's/###START_GENERATED_FILES###/###START_COMMENTED_GENERATED_FILES###/' $GITIGNORE
-        sed -i 's/###END_GENERATED_FILES###/###END_COMMENTED_GENERATED_FILES###/' $GITIGNORE
+        sed -i '' "$@"
     fi
-done
-
+}
 
+#### .gitignore processing ####
+for GITIGNORE in $(git ls-files --recurse-submodules -- '*.gitignore'); do
+        psed '/###START_GENERATED_FILES###/,/###END_GENERATED_FILES###/s/^/#/' "$GITIGNORE"
+        psed 's/###START_GENERATED_FILES###/###START_COMMENTED_GENERATED_FILES###/' "$GITIGNORE"
+        psed 's/###END_GENERATED_FILES###/###END_COMMENTED_GENERATED_FILES###/' "$GITIGNORE"
+done
 
 #### Build scripts ####
-
-# GEN_FILES defaults on (non-empty) in development, off (empty) in releases
-if [ -n "$unrelease" ]; then
-    r=' yes'
-else
-    r=''
-fi
-sed -i 's/^\(GEN_FILES[ ?:]*=\)\([^#]*\)/\1'"$r/" Makefile */Makefile
-
-# GEN_FILES defaults on in development, off in releases
-if [ -n "$unrelease" ]; then
-    r='ON'
-else
-    r='OFF'
-fi
-sed -i '/[Oo][Ff][Ff] in development/! s/^\( *option *( *GEN_FILES  *"[^"]*"  *\)\([A-Za-z0-9][A-Za-z0-9]*\)/\1'"$r/" CMakeLists.txt
+psed '/[Oo][Ff][Ff] in development/! s/^\( *GEN_FILES[ ?:]*= *\)\([A-Za-z0-9][A-Za-z0-9]*\)/\1OFF/' Makefile library/Makefile
+psed '/[Oo][Ff][Ff] in development/! s/^\( *option *( *GEN_FILES  *"[^"]*"  *\)\([A-Za-z0-9][A-Za-z0-9]*\)/\1OFF/' CMakeLists.txt tf-psa-crypto/CMakeLists.txt
```

Comment:
There's no `tf-psa-crypto` in 3.6.

```suggestion
psed '/[Oo][Ff][Ff] in development/! s/^\( *option *( *GEN_FILES  *"[^"]*"  *\)\([A-Za-z0-9][A-Za-z0-9]*\)/\1OFF/' CMakeLists.txt
```